# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerOpPromise.h
# Commit: 8fec60d0c499
# Full Hash: 8fec60d0c499fdf429f4d11da6bea0c65aecb1d4
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-27 09:40:54
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Depends on D138250
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r fb0ac558803d -r 8fec60d0c499 dom/serviceworkers/ServiceWorkerOpPromise.h
--- a/dom/serviceworkers/ServiceWorkerOpPromise.h	Sun Feb 27 00:26:21 2022 +0000
+++ b/dom/serviceworkers/ServiceWorkerOpPromise.h	Sun Feb 27 00:26:22 2022 +0000
@@ -30,10 +30,14 @@
 using FetchEventRespondWithPromise =
     MozPromise<FetchEventRespondWithResult, CancelInterceptionArgs, true>;
 
+using FetchEventPreloadResponseArgs =
+    Tuple<SafeRefPtr<InternalResponse>, IPCPerformanceTimingData, nsString,
+          nsString>;
+
 // The reject type int is arbitrary, since this promise will never get rejected.
 // Unfortunately void is not supported as a reject type.
 using FetchEventPreloadResponsePromise =
-    MozPromise<SafeRefPtr<InternalResponse>, int, true>;
+    MozPromise<FetchEventPreloadResponseArgs, int, true>;
 
 using ServiceWorkerOpPromise =
     MozPromise<ServiceWorkerOpResult, nsresult, true>;
